packages/cli/src/index.ts

```diff
@@ -82,6 +82,20 @@ function attachCommand<C extends Command<boolean>>(
   command: C,
   config: C extends DetachedCommand ? Config | undefined : Config,
 ): void {
+  // commander@9.x will internally push commands into an array structure!
+  // Commands with duplicate names (e.g. from config) must be reduced before
+  // calling this function.
+  // https://unpkg.com/browse/commander@9.4.1/lib/command.js#L1308
+  // @ts-ignore
+  if (program._findCommand(command.name)) {
```

Comment:
`_findCommand` is private. You might want to copy the implementation using public methods and properties rather than risk it changing.

Reply:
Definitely a small evil (note: `cmd._name` and `cmd._alias` are also private) — but I judged safe enough for this use (with a large comment).

Alternative would be tracking which commands we've added inside this module / via a wrapper.

Reply:
`cmd.name()` and `cmd.aliases()` are public. Both routines are setter/getter. With no argument, they return the current value.

Reply:
I see. Also checked we only have `name` in our input `Command` type, so this can be simplified 👍🏻